src/data/transforms/minmax.rs

```diff
@@ -130,28 +129,55 @@ impl<T: Float> Transformer<Matrix<T>> for MinMaxScaler<T> {
             .map(|(&(_, x), &s)| self.scaled_max - x * s)
             .collect::<Vec<_>>();
 
-        for row in inputs.iter_rows_mut() {
-            utils::in_place_vec_bin_op(row, &scales, |x, &y| {
-                *x = *x * y;
-            });
+        self.scale_factors = Some(Vector::new(scales));
+        self.const_factors = Some(Vector::new(consts));
+        Ok(())
+    }
 
-            utils::in_place_vec_bin_op(row, &consts, |x, &y| {
-                *x = *x + y;
-            });
+    fn transform(&mut self, mut inputs: Matrix<T>) -> Result<Matrix<T>, Error> {
+        match (&self.scale_factors, &self.const_factors) {
```

Comment:
I think using [`if let ...`](https://doc.rust-lang.org/book/if-let.html) would be a little clearer here.